lang/cs/Org.Apache.REEF.Client/Common/IApplicationReport.cs

```
@@ -0,0 +1,70 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using Org.Apache.REEF.Client.YARN.RestClient.DataModel;
+using Org.Apache.REEF.Utilities.Attributes;
+
+namespace Org.Apache.REEF.Client.Common
+{
+    /// <summary>
+    /// This interface represents application information mainated by Yarn RM.
+    /// This interface is modeled on Org.Apache.REEF.Client.YARN.RestClient.DataModel.Application. 
+    /// Documentation on the class Org.Apache.REEF.Client.YARN.RestClient.DataModel.Application
+    /// can be found here.
+    /// <a href="http://hadoop.apache.org/docs/r2.6.0/hadoop-yarn/hadoop-yarn-site/WebServicesIntro.html">
+    /// Hadoop RM REST API</a> documentation.
+    /// </summary>
+    [Unstable("0.16", "Working in progress. For local runtime, some of the property, such as FinalState and AppId are not implemented yet.")]
+    public interface IApplicationReport
+    {
+        /// <summary>
+        /// Get Yarn application name.
+        /// </summary>
+        string AppName { get; }
+        
+        /// <summary>
+        /// Get Yarn application start time.
+        /// </summary>
+        long StartedTime { get; }
+
+        /// <summary>
+        /// Get Yarn application finish time.
+        /// </summary>
+        long FinishedTime { get; }
+
+        /// <summary>
+        /// Get Yarn application number of running containers.
+        /// </summary>
+        int RunningContainers { get; }
+
+        /// <summary>
+        /// This method returns the url of http server running inside the driver.
+        /// e.g. http://hostname:port/
+        /// </summary>
+        string TrackingUrl { get; }
```

Yes, but wouldn't using `Url` make it type safe? It would allow things to fail early if a mal-formed string is returned.